services/app-api/utils/dynamo/dynamodb-lib.ts

```diff
@@ -55,6 +57,19 @@ export default {
 
     return items;
   },
+  scanAll: async (params: Omit<ScanCommandInput, "ExclusiveStartKey">) => {
+    let items: AnyObject[] = [];
+    let ExclusiveStartKey: Record<string, any> | undefined;
+
+    do {
+      const command = new ScanCommand({ ...params, ExclusiveStartKey });
+      const result = await client.send(command);
+      items = items.concat(result.Items ?? []);
+      ExclusiveStartKey = result.LastEvaluatedKey;
+    } while (ExclusiveStartKey);
```

Comment:
Non-blocking recommendation: The SDK's builtin `paginateScan` was made for just such an occasion.
```suggestion
    for await (let page of paginateScan({ client }, params)) {
      items = items.concat(page.Items ?? []);
    }
```

That would introduce a stylistic difference between `scanAll` and `queryAll`; if that bugs you then `paginateQuery` is also available.